staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/validator.go

```diff
@@ -114,25 +116,26 @@ func (a *validationActivation) Parent() interpreter.Activation {
 }
 
 // Compile compiles the cel expression defined in ValidatingAdmissionPolicy
-func (c *CELValidatorCompiler) Compile(p *v1alpha1.ValidatingAdmissionPolicy) Validator {
-	if len(p.Spec.Validations) == 0 {
+func (c *CELValidatorCompiler) Compile(failPolicy *v1.FailurePolicyType, validations []CELValidation, hasParam bool) Validator {
```

Comment:
Callers should be able to access whether there are compilation errors in order to raise known errors prior to execution.

Reply:
this latest version is capturing errors on the CompilationResult object which Evaluator uses to skip execution of the CEL expression